### ISSUE TITLE
Adding `size` information to artifact information of `.dir` files

### ISSUE THREAD

**Issue by csnick93** [iterative/dvc#9668]
Hi 👋 

First of all, thank you for your great work! We very much enjoy working with `dvc` in our AI team :) 

## Context

I am working with dvc with remote storage on GCS.

## Problem Description

I am missing the `size` attribute in the `<hash>.dir` files stored remotely for the individual artifacts belonging to the dvc tracked directory.
From what I have observed, the information stored in `.dvc` files is of the following structure:

```
outs:
- md5: <hash>
  size: <int>
  nfiles: <int>
  path: <str>
```

In case of a dvc directory, it's possible to get the information of the included individual dvc artifacts in the corresponding remotely stored `.dir` file. This file contains a list of entries, where each entry corresponds to an artifact tracked within the directory, with the information `md5` and `relpath`. Would it be possible to also add the `size` information here?

Happy to provide more information if needed! 

Thanks a lot for taking the time to read this :)

**Comment by pmrowla** [comment]
To clarify, are you looking for the size of the individual files in the directory (and not the total size of the directory, which is already listed in the `.dvc` file)?

**Comment by csnick93** [comment]
Hi @pmrowla,
exactly, I am looking for the size of the individual files. Thanks for pointing me to that issue, that should do 👍

**Comment by dberenbaum** [comment]
Hi @csnick93! I'd be curious to know more about how you plan to use that info and how you use DVC. Would be happy to jump on a call or discuss it here if you're willing to share more details.

**Comment by csnick93** [comment]
Hi @dberenbaum,
thanks for your interest, I'd be happy to share more details! I'll start by writing them here, but I'd also be up for a call.

Our goal is to sidestep `dvc` in our production environment when pulling the models/artifacts needed to run our production containers. The reason is that for our larger projects we have up to 65GB of artifacts to pull, which takes around 7 minutes with `dvc` to be completed. However, when using a mix of `gcloud storage cli` (really fast for large files) and `gsutil -m` (fast for lots of small files), we can get that down to around `1m40s`. Given your very transparent way of encoding where on the remote storage you save the files, it was pretty straightforward for me to implement that approach. However, one tedious part was when files were tracked as part of a directory. In that case I had to perform a bunch of extra calls to `GCS` to first know which files are part of the directory, and second I needed to query for each file its size (which is the more cumbersome/time consuming part). The reason I'd like to know the `size` is that by splitting the download based on the filesizes, I can achieve some more speedup. And that was my motivation to ask for the `size` attribute in the `.dir` files stored on the remote to save myself from having to query that myself every time.

Hope this makes it clearer! Let me know what you think and maybe you also have some thoughts on the general underlying approach of speeding up the `dvc pull` part. Also note, that in production I don't have the need for any kind of incremental pulling, I just want to pull the artifacts my `.dvc` files point to as quickly as possible.
